File: src/heap/lc_703_Kth_largest_element_in_stream/largest_element_in_stream.py
import heapq

class KthLargest:
    ### Min heap
    def __init__(self, k: int, nums: list[int]):
        self.nums = nums
        self.k = k
        heapq.heapify(self.nums)
        while len(self.nums) > k:
            heapq.heappop(self.nums)
        
    def add(self, val: int) -> int:
        heapq.heappush(self.nums, val)
        if self.k < len(self.nums):
            heapq.heappop(self.nums)
        return self.nums[0]
        

File: src/heap/lc_703_Kth_largest_element_in_stream/largest_element_in_stream.java
import java.util.*;
class KthLargest {

    final PriorityQueue<Integer> pq = new PriorityQueue(); //min heap
    final int k;

    public KthLargest(int k, int[] nums) {
        this.k = k;
        for(int i = 0; i < nums.length; i++) {
            pq.offer(nums[i]);
        }
        while (pq.size() > k) {
            pq.remove();
        }
    }
    
    public int add(int val) {
        pq.offer(val);
        if(this.k < pq.size())
            pq.remove();
        int v = pq.remove();
        pq.offer(v);
        return v;
    }
}